File: backend/src/main/java/proyecto/service/OrdenService.java
package proyecto.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import proyecto.entities.OrdenEntity;
import proyecto.repositories.ClienteRepository;
import proyecto.repositories.OrdenRepository;

import java.util.List;

@Service
public class OrdenService {

    @Autowired
    OrdenRepository ordenRepository;

    @Autowired
    ClienteRepository clienteRepository;

    public List<OrdenEntity> findAll(){
        return ordenRepository.findAll();
    }

    public OrdenEntity findById(int id_orden){
        return ordenRepository.findById(id_orden);
    }

    public OrdenEntity create(OrdenEntity orden){
        return ordenRepository.create(orden);
    }

    public OrdenEntity update(OrdenEntity orden){
        return ordenRepository.update(orden);
    }

    public boolean delete(int id_orden){
        return ordenRepository.delete(id_orden);
    }

    public List<OrdenEntity> findByCliente(int id_cliente){
        return ordenRepository.findByCliente(id_cliente);
    }
}

File: front-end/src/services/ordenService.js
import httpCommon from "@/http-common";

const getAll = () => {
  return httpCommon.get("/api/orden");
}

const get = id => {
    return httpCommon.get(`/api/orden/${id}`);
}

const create = data => {
    return httpCommon.post("/api/orden/create", data);
}

const update = data => {
    return httpCommon.put("/api/orden/update", data);
}

findByCliente = idCliente => {
    return httpCommon.get(`/api/orden/findByCliente/${idCliente}`);
}

export default {
    getAll,
    get,
    create,
    update,
    findByCliente
};